### PR TITLE
Added admin routes

### PR DESCRIPTION
**Pull request by pal-16**
The commit includes routes for (verified via postman)

- Creating Product
- Editing Product
- Deleting Product
- Retrieving Users

Kindly let me know if any changes to be made.

### CODE REVIEW

**Comment by pal-16** [comment]
Middleware left to be added.

File: server/src/controllers/admin.controller.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      return res.status(200).json({
+        product
+      });
+    } catch (err) {
+      return next({
+        status: 400,
+        message: err.message
+      });
+    }
+  },
```
Comment (BhavyaSheth22) [inline server/src/controllers/admin.controller.ts:24]: Where is this next function passing parameters to? If not required, I think we can remove them from all the controller functions

File: server/tsconfig.json [hunk 2]
```diff
@@ -2,12 +2,17 @@
...
+      "es6",
+      "es2017",
+      "esnext.asynciterable"
+    ],
     "sourceMap": true,
     "outDir": "./compiled",
     "moduleResolution": "node",
     "removeComments": true,
-    "noImplicitAny": true,
+    "noImplicitAny": false,
```
Comment (BhavyaSheth22) [inline server/tsconfig.json:15]: Please change this to true once type checks for next are added.

**Review by BhavyaSheth22 — CHANGES_REQUESTED** [review]
Thanks for this! Please add the isAdmin middleware. Apart from that, just a few changes:

File: server/src/controllers/admin.controller.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import User from "../models/User/User";
+import Product from "../models/Product/Product";
+
+import express from "express";
+
+const AdminController = {
+  async newProduct(
+    req: express.Request,
+    res: express.Response,
+    next
```
[the remark this answers is not in the document]
Comment (pal-16) [reply server/src/controllers/admin.controller.ts:10]: Yes, I have removed that and added a status code of 500 in case of any error caught.

**Review by ShubhankarKG — CHANGES_REQUESTED** [review]
Thanks, when using `async/ await`, there's no need for promises again, and vice versa.

**Review by ShubhankarKG — APPROVED** [review]

**Review by pal-16 — COMMENTED** [review]
I am refactoring the routes and removing the extra function. Thanks for the edit.

**Review by BhavyaSheth22 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**